src/bgen/StringExtensions.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections;
+
+#nullable enable
+
+// Fixes bug 27430 - btouch doesn't escape identifiers with the same name as C# keywords
+public static class StringExtensions {
+
+	public static string Quote (this string? self)
+	{
+		return self switch {
+			null => String.Empty,
+			"" => @"""""",
+			_ => $"@\"{self.Replace ("\"", "\"\"")}\""
+		};
+	}
+
+	public static string CamelCase (this string ins)
+	{
+		return Char.ToUpper (ins [0]) + ins.Substring (1);
+	}
+
+	public static string PascalCase (this string ins)
+	{
+		return Char.ToLower (ins [0]) + ins.Substring (1);
+	}
+
+	public static string Capitalize (this string str)
+	{
+		if (str.StartsWith ("@", StringComparison.Ordinal))
+			return char.ToUpper (str [1]) + str.Substring (2);
+
+		return char.ToUpper (str [0]) + str.Substring (1);
+	}
+
+	public static string? GetSafeParamName (this string? paramName)
+	{
+		if (paramName is null)
+			return paramName;
+
+		if (!IsValidIdentifier (paramName, out var hasIllegalChars)) {
+			return hasIllegalChars ? null : "@" + paramName;
+		}
+		return paramName;
+	}
+
+	// Since we're building against the iOS assemblies and there's no code generation there,
+	// I'm bringing the implementation from:
+	// mono/mcs/class//System/Microsoft.CSharp/CSharpCodeGenerator.cs
+	static bool IsValidIdentifier (string? identifier, out bool hasIllegalChars)
+	{
+		hasIllegalChars = false;
+		if (identifier is null || identifier.Length == 0)
```

Comment:
```suggestion
		if (String.IsNullOrEmpty(identifier))
```
Might fit a little better here, but no big deal :)